gpii/node_modules/solutionsRegistry/src/js/filtered-settings-middleware.js

```diff
@@ -0,0 +1,174 @@
+                                settings: "{that}.options.filteredSettingsSchema"
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+});
+
+fluid.defaults("gpii.universal.solutionsRegistry.kettle.request.http.preferences", {
+    gradeNames: ["gpii.universal.solutionsRegistry.kettle.request.http.base"],
+    rules: {
+        filterInput: {
+            "": {
+                transform: {
+                    type: "fluid.transforms.firstValue",
+                    // TODO: discuss how to handle contexts other than gpii-default.
+                    values: [
+                        "body.contexts.gpii-default.preferences.application",
+                        "body.contexts.gpii-default.preferences"
+                    ]
+                }
+            }
+        }
+    },
+    inputSchema: {
+        type: "object",
+        properties: {
+            body: {
+                type: "object",
+                properties: {
+                    contexts: {
+                        type: "object",
+                        required: true,
+                        minProperties: 1,
+                        additionalProperties: {
+                            type: "object",
+                            properties: {
+                                metadata: {
+                                    type: "array",
+                                    items: {
+                                        type: "object",
+                                        properties: {
+                                            scope: {
+                                                type: "array",
+                                                items: {
+                                                    oneOf: [
+                                                        { type: "string" },
+                                                        { type: "array", items: { type: "string" } }
+                                                    ]
+                                                }
+                                            },
+                                            source: {
+                                                type: "string"
+                                            },
+                                            type: {
+                                                type: "string"
+                                            }
+                                        }
+                                    }
+                                },
+                                name: {
+                                    type: "string",
+                                    required: true
+                                },
+                                preferences: {
+                                    required: true,
+                                    anyOf: [
+                                        // Without a "view"
```

Comment:
Well, this is awkward! It seems like we can't actually validate this payload adequately since we can't constrain the schema contents to be the ones which depend on whether a view has been provided or not

Reply:
We certainly can if we start using the `if`, `then` and `else` keywords.

Reply:
I imagine that is probably not something we do want to do, but the alternatives aren't entirely clear. The current approach will be fine to be merged but we should write up our thinking on improvements.

Reply:
I left a TODO comment above this spot with thoughts on how this might be accomplished.